pkg/api/types.go

```diff
@@ -455,6 +455,10 @@ type TestStepConfiguration struct {
 	// Commands are the shell commands to run in
 	// the repository root to execute tests.
 	Commands string `json:"commands,omitempty"`
+
```

Comment:
You want these in `TestStep` and `LiteralTestStep` - not here.

Reply:
As of `a67ec4ba6`, I'm touching two streams:

a. `api.TestStepConfiguration` -> `steps.TestStep` -> `steps.PodStepConfiguration` -> `podStep.generatePodForStep`.
b. `api.LiteralTestStep` -> `multiStageTestStep.generatePods`.

(b) makes sense to me for multi-step ref YAML.  I'm not really sure what (a) is about, or how (a) is related to (b).

Reply:
`TestStepConfiguration` corresponds to the entries in the `tests` section of the configuration.  It will affect all types of tests (container, template, multi-stage).

Reply:
Is that ok?  Or do we want to exclude those for now?

Reply:
I think we want to have this on `TestStep` level, not `TestStepConfiguration` for consistency with `LiteralTestStep` (so that YAML indent levels are consistent between the two). We'll want people to mostly use test steps anyway.

Reply:
[`TestStep` contains a `LiteralTestStep`][1], so if we add these there, you'd have the possibility of disagreement between `TestStep.ActiveDeadlineSeconds` and `TestStep.LiteralTestStep.ActiveDeadlineSeconds`, right?  My current positioning has the new properties as siblings of `commands` in two of the three properties that have `Commands` (`PipelineImageCacheStepConfiguration` is the third type with `Commands`, and I dunno if we want the timeout properties in there or not).  Do you still want me to shift some of these properties to a different struct?

[1]: https://github.com/openshift/ci-tools/blob/c450a3632d5c7b4dcc02c37063d0a027840338bc/pkg/api/types.go#L633-L634

Reply:
No, there can be no disagreement. These fields should not be on `TestStepConfiguration`.

Reply:
Dropped from `TestStepConfiguration` with 35e4b8e96 -> e3eb04db4.  So now the only `pkg/api/types.go` is `LiteralTestStep`.  Did you want them on `TestStep` too, per [this earlier comment][1], or does [my argument][2] that `TestStep` contains `LiteralTestStep` mean that touching just `LiteralTestStep` is sufficient?

[1]: https://github.com/openshift/ci-tools/pull/1257#discussion_r494004151
[2]: https://github.com/openshift/ci-tools/pull/1257#discussion_r496203335